src/wui/mapview.cc

```diff
@@ -587,10 +586,61 @@ Widelands::NodeAndTriangle<> MapView::track_sel(const Vector2i& p) {
 	return node_and_triangle;
 }
 
+bool MapView::scroll_map() {
```

Comment:
There are lots of similar-looking lines in this function. How about simplifying it like:
```
int32_t distance_to_scroll_x = 0;
int32_t distance_to_scroll_y = 0;
if (…left key down…) distance_to_scroll_x -=scrollDistanceX;
if (…right key down…) distance_to_scroll_x +=scrollDistanceX;
if (…up key down…) distance_to_scroll_y -=scrollDistanceY;
if (…down key down…) distance_to_scroll_y +=scrollDistanceY;
if (distance_to_scroll_x == 0 && distance_to_scroll_y == 0) return false;
pan_by(…distance_to_scroll_x, distance_to_scroll_y…)
return true
```
Also please use under_scores instead of CamelCase for the names of `ScrollDistanceX/Y` and make them `const`, and cache the value of `!(SDL_GetModState() & KMOD_NUM)` in a `const bool`.